### ISSUE TITLE
Fatal error for lorem sentences and paragraphs

### ISSUE THREAD

**Issue by merchako** [tillkruss/alfred-lorem-ipsum#8]
I'm getting fatal errors when I run the workflow with its default values `lorem paragraphs` and `lorem sentences`, but `lorem words` workds fine.

```
[19:28:37.856] Lorem Ipsum[Script Filter] Queuing argument '1'
[19:28:37.975] Lorem Ipsum[Script Filter] Script with argv '(null)' finished
[19:28:37.987] STDERR: Lorem Ipsum[Script Filter] PHP Deprecated:  Implicit conversion from float 29.492137920308757 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138
PHP Deprecated:  Implicit conversion from float 24.16696938580465 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138
PHP Deprecated:  Implicit conversion from float 24.623126871658936 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138
[19:28:37.989] Lorem Ipsum[Script Filter] Deprecated: Implicit conversion from float 29.492137920308757 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138

Deprecated: Implicit conversion from float 24.16696938580465 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138

Deprecated: Implicit conversion from float 24.623126871658936 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138
{"items":[{"arg":"Laoreet viverra himenaeos nostra bibendum molestie sagittis morbi ornare netus id, per condimentum cubilia cursus porta consequat nisl montes felis, dapibus nec ligula torquent donec quisque pretium justo posuere. Egestas curabitur tempor malesuada laoreet facilisi senectus faucibus vehicula dictum ipsum ligula pretium, rhoncus ridiculus sagittis cubilia dignissim morbi commodo posuere dictumst luctus auctor. Id imperdiet phasellus netus sodales quis lectus dui tortor, ornare pretium nunc placerat parturient eros eget volutpat, eu morbi bibendum vestibulum montes congue magnis.","subtitle":"Generate 1 paragraph...","text":{"copy":"Laoreet viverra himenaeos nostra bibendum molestie sagittis morbi ornare netus id, per condimentum cubilia cursus porta consequat nisl montes felis, dapibus nec ligula torquent donec quisque pretium justo posuere. Egestas curabitur tempor malesuada laoreet facilisi senectus faucibus vehicula dictum ipsum ligula pretium, rhoncus ridiculus sagittis cubilia dignissim morbi commodo posuere dictumst luctus auctor. Id imperdiet phasellus netus sodales quis lectus dui tortor, ornare pretium nunc placerat parturient eros eget volutpat, eu morbi bibendum vestibulum montes congue magnis.","largetype":"Laoreet viverra himenaeos nostra bibendum molestie sagittis morbi ornare netus id, per condimentum cubilia cursus porta consequat nisl montes felis, dapibus nec ligula torquent donec quisque pretium justo posuere. Egestas curabitur tempor malesuada laoreet facilisi senectus faucibus vehicula dictum ipsum ligula pretium, rhoncus ridiculus sagittis cubilia dignissim morbi commodo posuere dictumst luctus auctor. Id imperdiet phasellus netus sodales quis lectus dui tortor, ornare pretium nunc placerat parturient eros eget volutpat, eu morbi bibendum vestibulum montes congue magnis."},"title":"Paragraphs","uid":"lorem-words","valid":true}]}
[19:28:37.991] ERROR: Lorem Ipsum[Script Filter] JSON error: JSON text did not start with array or object and option to allow fragments not set. around line 1, column 0. in JSON:
Deprecated: Implicit conversion from float 29.492137920308757 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138

Deprecated: Implicit conversion from float 24.16696938580465 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138

Deprecated: Implicit conversion from float 24.623126871658936 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138
{"items":[{"arg":"Laoreet viverra himenaeos nostra bibendum molestie sagittis morbi ornare netus id, per condimentum cubilia cursus porta consequat nisl montes felis, dapibus nec ligula torquent donec quisque pretium justo posuere. Egestas curabitur tempor malesuada laoreet facilisi senectus faucibus vehicula dictum ipsum ligula pretium, rhoncus ridiculus sagittis cubilia dignissim morbi commodo posuere dictumst luctus auctor. Id imperdiet phasellus netus sodales quis lectus dui tortor, ornare pretium nunc placerat parturient eros eget volutpat, eu morbi bibendum vestibulum montes congue magnis.","subtitle":"Generate 1 paragraph...","text":{"copy":"Laoreet viverra himenaeos nostra bibendum molestie sagittis morbi ornare netus id, per condimentum cubilia cursus porta consequat nisl montes felis, dapibus nec ligula torquent donec quisque pretium justo posuere. Egestas curabitur tempor malesuada laoreet facilisi senectus faucibus vehicula dictum ipsum ligula pretium, rhoncus ridiculus sagittis cubilia dignissim morbi commodo posuere dictumst luctus auctor. Id imperdiet phasellus netus sodales quis lectus dui tortor, ornare pretium nunc placerat parturient eros eget volutpat, eu morbi bibendum vestibulum montes congue magnis.","largetype":"Laoreet viverra himenaeos nostra bibendum molestie sagittis morbi ornare netus id, per condimentum cubilia cursus porta consequat nisl montes felis, dapibus nec ligula torquent donec quisque pretium justo posuere. Egestas curabitur tempor malesuada laoreet facilisi senectus faucibus vehicula dictum ipsum ligula pretium, rhoncus ridiculus sagittis cubilia dignissim morbi commodo posuere dictumst luctus auctor. Id imperdiet phasellus netus sodales quis lectus dui tortor, ornare pretium nunc placerat parturient eros eget volutpat, eu morbi bibendum vestibulum montes congue magnis."},"title":"Paragraphs","uid":"lorem-words","valid":true}]}
[19:28:45.644] Lorem Ipsum[Script Filter] Queuing argument '1'
[19:28:45.738] Lorem Ipsum[Script Filter] Script with argv '(null)' finished
[19:28:45.749] STDERR: Lorem Ipsum[Script Filter] PHP Deprecated:  Implicit conversion from float 19.424283288804375 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138
[19:28:45.750] Lorem Ipsum[Script Filter] Deprecated: Implicit conversion from float 19.424283288804375 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138
{"items":[{"arg":"Est class leo velit fusce maecenas massa, per dui scelerisque sodales a aptent interdum, molestie maximus viverra aliquam faucibus.","subtitle":"Generate 1 sentence...","text":{"copy":"Est class leo velit fusce maecenas massa, per dui scelerisque sodales a aptent interdum, molestie maximus viverra aliquam faucibus.","largetype":"Est class leo velit fusce maecenas massa, per dui scelerisque sodales a aptent interdum, molestie maximus viverra aliquam faucibus."},"title":"Sentences","uid":"lorem-words","valid":true}]}
[19:28:45.750] ERROR: Lorem Ipsum[Script Filter] JSON error: JSON text did not start with array or object and option to allow fragments not set. around line 1, column 0. in JSON:
Deprecated: Implicit conversion from float 19.424283288804375 to int loses precision in /Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php on line 138
{"items":[{"arg":"Est class leo velit fusce maecenas massa, per dui scelerisque sodales a aptent interdum, molestie maximus viverra aliquam faucibus.","subtitle":"Generate 1 sentence...","text":{"copy":"Est class leo velit fusce maecenas massa, per dui scelerisque sodales a aptent interdum, molestie maximus viverra aliquam faucibus.","largetype":"Est class leo velit fusce maecenas massa, per dui scelerisque sodales a aptent interdum, molestie maximus viverra aliquam faucibus."},"title":"Sentences","uid":"lorem-words","valid":true}]}
```
![Screen Shot 2022-01-05 at 19 30 49](https://user-images.githubusercontent.com/441789/148313570-f4f16f9c-9eb4-43d5-b265-ce22336134a8.png)

I've reinstall twice and recently installed PHP with Homebrew. I've come to the end what I know to do. Help? 

—
Alfred 4.6.1
macOS 12.0.1   Monterey
/opt/homebrew/bin/php --version > PHP 8.1.1

**Comment by rverrips** [comment]
Dear @merchako 

Thank you for logging this issue and providing the detailed logs and environment specifics. 

The issue is related to the third party library joshtronic/php-loremipsum which is not (yet) compatible with PHP 8.1 due to a deprecation that was introduced with that version of php resulting the error you see in your logs. 

I have already submitted [a pull request to the library](https://github.com/joshtronic/php-loremipsum/pull/14) with the hope of addressing this.   Once that PR is merged @tillkruss should be able to create a new version of the Alfred Workflow using the updated library which will then make the Alfred Workflow compatible with PHP 8.1 

In the interim I would suggest you downgrade to PHP 8.0 

Alternatively, you could also manually try the fix to the library in your instance - update this file:
/Users/merc/Dropbox/Alfred Preferences/Alfred.alfredpreferences/workflows/user.workflow.011387CF-B378-4C19-8B59-29E5079C05CE/vendor/joshtronic/php-loremipsum/src/LoremIpsum.php

Look for this section (around line 128):
```php
public function words($count = 1, $tags = false, $array = false)
    {
        $words      = array();
        $word_count = 0;

        // Shuffles and appends the word list to compensate for count
```
Add the conversion of $count to an integer
```php
public function words($count = 1, $tags = false, $array = false)
    {
        $count      = (int) $count;
        $words      = array();
        $word_count = 0;

        // Shuffles and appends the word list to compensate for count
```
Please let us know which solution you try and if your issue is resolved.

**Comment by merchako** [comment]
Thank you, @rverrips. I manually fixed the library in my instance by updating LoremIpsum.php as you suggested. It worked like a charm!

**Comment by tillkruss** [comment]
Let's wait for https://github.com/joshtronic/php-loremipsum/pull/14

**Comment by rverrips** [comment]
@tillkruss 

https://github.com/joshtronic/php-loremipsum/pull/14 has been merged and a new release ([2.1.0](https://github.com/joshtronic/php-loremipsum/releases/tag/2.1.0)) has been tagged.

**Comment by tillkruss** [comment]
Tagged [v0.2.2](https://github.com/tillkruss/alfred-lorem-ipsum/releases/tag/v0.2.2).
